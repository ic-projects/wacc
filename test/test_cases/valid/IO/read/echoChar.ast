-- Test: echoChar.wacc

-- Uploaded file: 
---------------------------------------------------------------
# echo the user's input char

# Output:
# enter a character to echo
# #input#
# #output#

# Program:

begin
	char c = '\0' ;
  println "enter a character to echo";
	read c ;
	println c 
end
---------------------------------------------------------------

-- Compiler Output:
-- Compiling...
-- Printing AST...
echoChar.ast contents are:
===========================================================
0	Program
1	  - int main()
2	    - DECLARE
3	      - TYPE
4	        - char
5	      - LHS
6	        - c
7	      - RHS
8	        - '\0'
9	    - PRINTLN
10	      - "enter a character to echo"
11	    - READ
12	      - c
13	    - PRINTLN
14	      - c
===========================================================
-- Finished

