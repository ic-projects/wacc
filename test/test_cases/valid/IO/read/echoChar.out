-- Compiling...
-- Assembling and Linking...
-- Executing...
===========================================================
enter a character to echo
